aspnetcore/performance/rate-limit.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Rate limiting middleware in ASP.NET Core
+author: rick-anderson
+description: Learn how limit requests in ASP.NET Core apps
+ms.author: riande
+ms.custom: mvc
+ms.date: 8/28/2022
+uid: performance/rate-limit
+---
+
+# Rate limiting middleware in ASP.NET Core
+
+By [Arvin Kahbazi](https://github.com/Kahbazi) and [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+:::moniker range=">= aspnetcore-7.0"
+
+The `Microsoft.AspNetCore.RateLimiting` middleware provides rate limiting middleware. Apps configure rate limiting policies and then attach the policies to endpoints. Apps using rate limiting should be carefully load tested and reviewed before deploying. See [Testing endpoints with rate limiting](#test7) in this article for more information.
+
+## Rate limiter algorithms
+
+The [`RateLimiterOptionsExtensions`](/dotnet/api/microsoft.aspnetcore.ratelimiting.ratelimiteroptionsextensions) class provides the following extension methods for rate limiting:
+
+* [Fixed window](#fixed)
+* [Sliding window](#slide)
+* [Token bucket](#token)
+* [Concurrency](#concur)
+
+<a name="fixed"></a>
+
+### Fixed window limiter
+
+The [`AddFixedWindowLimiter`](/dotnet/api/microsoft.aspnetcore.ratelimiting.ratelimiteroptionsextensions.addfixedwindowlimiter#microsoft-aspnetcore-ratelimiting-ratelimiteroptionsextensions-addfixedwindowlimiter(microsoft-aspnetcore-ratelimiting-ratelimiteroptions-system-string-system-threading-ratelimiting-fixedwindowratelimiteroptions)) method uses a fixed time window to limit requests. When the time window expires, a new time window starts and the request limit is reset.
+
+Consider the following code:
+
+:::code language="csharp" source="~/../AspNetCore.Docs.Samples/fundamentals/middleware/rate-limit/WebRateLimitAuth/Program.cs" id="snippet_fixed":::
+
+The preceding code:
+* Calls [UseRateLimiter](/dotnet/api/microsoft.aspnetcore.builder.ratelimiterapplicationbuilderextensions.useratelimiter) to enable rate limiting.
+* Creates a fixed window limiter with a policy name of `"fixed"` and sets:
+* `permitLimit` to 4 and the time `window` to 12. A maximum of 4 requests per each 12-second window are allowed.
```

Comment:
Would be easier to read if these bullets were indented one further

Reply:
@wtgodbe  I don't understand. This is the Microsoft docs standard indenting. I don't see how double indenting would make it easier to read.

<img width="335" alt="image" src="https://user-images.githubusercontent.com/3605364/186993017-841c71d9-ef29-468d-b9b8-8769035c4d4a.png">